src/main/java/com/iexec/blockchain/broker/BrokerService.java

```diff
@@ -115,41 +112,81 @@ private Optional<String> fireMatchOrders(BrokerOrder brokerOrder) {
+                );
+            }
+            if (!isAccepted) {
+                log.error("Prices not accepted (too expensive) {}", messageDetails);
+            }
+            return isAccepted;
+        } catch (NullPointerException e) {
+            log.error("Failed to check hasRequesterAcceptedPrices", e);
             return false;
         }
-        boolean isAppPriceAccepted = requestOrder.getAppmaxprice().longValue() >= appPrice.longValue();
-        boolean isPoolPriceAccepted = requestOrder.getWorkerpoolmaxprice().longValue() >= workerpoolPrice.longValue();
-        boolean isDatasetPriceAccepted = requestOrder.getDatasetmaxprice().longValue() >= datasetPrice.longValue();
-        boolean isAccepted = isAppPriceAccepted && isPoolPriceAccepted && isDatasetPriceAccepted;
-        if (!isAccepted) {
-            log.error("Prices not accepted (too expensive) [isAppPriceAccepted:{}, " +
-                    "isPoolPriceAccepted:{}]", isAppPriceAccepted, isPoolPriceAccepted);
-        }
-        return isAccepted;
     }
 
-    private boolean hasRequesterDepositedEnough(RequestOrder requestOrder, long deposit) {
-        if (requestOrder == null || requestOrder.getWorkerpoolmaxprice() == null
-                || requestOrder.getAppmaxprice() == null) {
-            log.error("Failed to check hasRequesterDepositedEnough (null requestOrder)");
-            return false;
-        }
-        long price = requestOrder.getWorkerpoolmaxprice().add(requestOrder.getAppmaxprice()).add(requestOrder.getDatasetmaxprice()).longValue();
-        if (price > deposit) {
-            log.error("Deposit too low [price:{}, deposit:{}]", price, deposit);
+    boolean hasRequesterDepositedEnough(RequestOrder requestOrder, long deposit, boolean withDataset) {
+        try {
+            checkRequestOrder(requestOrder, withDataset);
+
+            BigInteger price = requestOrder.getWorkerpoolmaxprice().add(requestOrder.getAppmaxprice());
+            if (withDataset) {
+                price = price.add(requestOrder.getDatasetmaxprice());
+            }
+            if (price.longValue() > deposit) {
+                log.error("Deposit too low [price:{}, deposit:{}]", price, deposit);
+                return false;
+            }
+            return true;
+        } catch (NullPointerException e) {
+            log.error("Failed to check hasRequesterDepositedEnough", e);
             return false;
         }
-        return true;
     }
+
+    /**
+     * Checks if a dataset is part of the order.
+     * A valid request requires a dataset address which is not empty and not the empty address.
+     *
+     * @param datasetAddress Dataset address to check
+     * @return true if a dataset is part of the request, false otherwise.
+     */
+     boolean withDataset(String datasetAddress) {
+        return StringUtils.isNotEmpty(datasetAddress) && !datasetAddress.equals(BytesUtils.EMPTY_ADDRESS);
```

Comment:
I think you can use https://github.com/iExecBlockchainComputing/iexec-common/blob/master/src/main/java/com/iexec/common/utils/EthAddress.java#L30

Reply:
This is a really interesting point. I just don't know if we do this right now or if we try to improve the model to manipulate `EthAddress` in orders instead of plain `String`.

Reply:
Kept as is for the moment. A more thorough study is required on classes available in web3j library.

Reply:
Sure :+1: